src/ui-components/input-field/input-field.spec.js

```diff
@@ -0,0 +1,315 @@
+        const wrapper = shallow(<InputField validationMessages={testValidationMessages} />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeTruthy();
+        expect(wrapper.find(`.${CSS_CLASSES.VALIDATION_MESSAGE}`).length).toEqual(3);
+      });
+      it("are not rendered when not provided", () => {
+        const wrapper = shallow(<InputField />);
+
+        expect(wrapper.exists(`.${CSS_CLASSES.VALIDATION_MESSAGE}`)).toBeFalsy();
+      });
+    });
+
+    describe("label", () => {
+      it("is rendered when provided", () => {
+        const testLabel = "this is label";
+        const wrapper = shallow(<InputField label={testLabel} />);
+
+        expect(wrapper.exists("label")).toBeTruthy();
+        expect(wrapper.find("label").text().trim()).toEqual(testLabel);
+      });
+      it("is not rendered when not provided", () => {
+        const wrapper = shallow(<InputField />);
+
+        expect(wrapper.exists("label")).toBeFalsy();
+      });
+      it("has label lowered class applied when field is empty", () => {
+        const testLabel = "this is label";
+        const wrapper = shallow(<InputField label={testLabel} />);
+
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_LOWERED)).toEqual(true);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_RAISED)).toEqual(false);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_FOCUSED)).toEqual(false);
+      });
+      it("has label raised class applied when field is not empty", () => {
+        const testLabel = "this is label";
+        const testValue = "this is value"
+        const wrapper = shallow(<InputField label={testLabel} value={testValue} />);
+
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_RAISED)).toEqual(true);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_LOWERED)).toEqual(false);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_FOCUSED)).toEqual(false);
+      });
+      it("has label focused class applied when field is focused", () => {
+        const testLabel = "this is label";
+        const wrapper = shallow(<InputField label={testLabel} />);
+        const mockEvent = {
+          stopPropagation: () => {}
+        };
+
+        wrapper.find("input").simulate("focus", mockEvent);
+
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_FOCUSED)).toEqual(true);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_RAISED)).toEqual(true);
+        expect(wrapper.find("label").hasClass(CSS_CLASSES.LABEL_LOWERED)).toEqual(false);
+      });
+    });
+
+    it("onChange event changes state.value", () => {
+      const wrapper = shallow(<InputField />);
+      const mockEvent = {
+        target: { value: "test text"},
+        stopPropagation: () => {}
+      };
+
+      wrapper.find("input").simulate("change", mockEvent);
+      
+      expect(wrapper.state().value).toEqual("test text");

```

Comment:
Here you could reuse the `mockEvent.target.value`.

- [ ] Reuse mockEvent or refute